### PR TITLE
Basic Acceptance Testing

### PR DESCRIPTION
**Pull request by mumby0168**
Related #164 

This is the general idea of what I see the acceptance tests to look like, thoughts @IEvangelist?

There are a few points for discussion below:

1. How often do we run these? (nightly, for every PR? for every merge to master?), I feel there are trade off's with each approach
2. Where do we set up the cosmos instance to test against? (The free tier of 1000 RUs should keep this FREE.
3. We need to find a way to exclude these tests being run for every case, this should be opt-in only I think.

I have added an example test that gives an idea that these tests are not just to test a single `CreateAsync` for example but to test the library doing realistic sets of operations and working as expected.

### CODE REVIEW

File: tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public abstract class CosmosRepositoryAcceptanceTest
+{
+    private const string ProductsInfoContainer = "products-info";
+    private const string DefaultPartitionKey = "/partitionKey";
+    protected const string TechnologyCategoryId = "Techonology";
+
+    protected readonly IRepository<Product> _productsRepository;
+    protected readonly IRepository<Rating> _ratingsRepository;
+
+    protected EquivalencyAssertionOptions<Product> DefaultProductEquivalencyOptions(EquivalencyAssertionOptions<Product> options)
```
Comment (IEvangelist) [inline tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs:23]:
```suggestion
    protected EquivalencyAssertionOptions<Product> DefaultProductEquivalencyOptions(
        EquivalencyAssertionOptions<Product> options)
```

File: tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+
+    protected CosmosRepositoryAcceptanceTest(Action<RepositoryOptions> builderOptions)
+    {
+        ConfigurationBuilder config = new();
+        config.AddUserSecrets<CosmosRepositoryAcceptanceTest>();
+
+        ServiceCollection services = new();
+
```
Comment (IEvangelist) [inline tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs:40]:
```suggestion
```

File: tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    protected EquivalencyAssertionOptions<Product> DefaultProductEquivalencyOptions(EquivalencyAssertionOptions<Product> options)
+    {
+        options.Excluding(x => x.Etag);
+        options.Excluding(x => x.CreatedTimeUtc);
+        options.Excluding(x => x.LastUpdatedTimeRaw);
+        options.Excluding(x => x.LastUpdatedTimeUtc);
+
+        return options;
+    }
+
```
Comment (IEvangelist) [inline tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/CosmosRepositoryAcceptanceTest.cs:32]:
```suggestion
```

**Review by IEvangelist — APPROVED** [review]
I like it, clean and simple.

**Comment by IEvangelist** [comment]
> How often do we run these? (nightly, for every PR? for every merge to master?), I feel there are trade off's with each approach

As for the number of times they run, we could start by running them `on` every `push` to `main`. And if that's too much, we can reassess and tune it back as needed.

> We need to find a way to exclude these tests being run for every case, this should be opt-in only I think.

We use xUnit, we should be able to use categories.

> Where do we set up the cosmos instance to test against? (The free tier of 1000 RUs should keep this FREE.

I'll reach out to the Cosmos team and see if they might be able to help us out with a test account for this.

**Comment by mumby0168** [comment]
> > How often do we run these? (nightly, for every PR? for every merge to master?), I feel there are trade off's with each approach
> 
> As for the number of times they run, we could start by running them `on` every `push` to `main`. And if that's too much, we can reassess and tune it back as needed.
> 
> > We need to find a way to exclude these tests being run for every case, this should be opt-in only I think.
> 
> We use xUnit, we should be able to use categories.
> 
> > Where do we set up the cosmos instance to test against? (The free tier of 1000 RUs should keep this FREE.
> 
> I'll reach out to the Cosmos team and see if they might be able to help us out with a test account for this.

Okay that sound's great, let me know what the outcome is 😃

I'll work on adding the xUnit categories to this.

I agree let's run it on master only for now and then we can add a manual run for this if we think a PR branch should be checked before a merge.

**Review by IEvangelist — DISMISSED** [review]

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**